File: timing1.py
"""
File: timing1.py in page 38
Prints the running times fro problem size that double,
using a single loop.
"""
import time 

problem_size = 10000000

print("%12s%16s" % ("Problem Size", "Seconds"))

for count in range(5):
    start = time.time()
    work = 1 
    for work in range(problem_size):
        work += 1
        work -= 1

    elapsed = time.time() - start

    print("%12d%16.3f" % (problem_size, elapsed))
    problem_size *= 2

File: timing1.c
/* File: timing.c in page 38
 * Prints the running times for problem sizes that double,
 * using a single loop
 */
#include <stdio.h>
#include <time.h>

int main(void)
{
        int problem_size = 10000000; 
        int times, work;
        clock_t start, end;

        printf("%12s%16s\n", "Problem Size", "Seconds");
        
        for (times = 0; times < 5; times++)
        {
                start = clock(); 
                
                for (work = 0; work <= problem_size;work++)
                {                        
                        work += 1;
                        work -= 1;
                }
                end = clock(); 
		
		double elapsed = (double)(end - start) / CLOCKS_PER_SEC;
                printf("%12d%16.3f\n", problem_size, elapsed);

                problem_size *= 2;
        }

        return 0;
} 
